src/python/pants/backend/shell/goals/test_test.py

```diff
@@ -78,6 +83,24 @@ def test_shell_command_as_test(rule_runner: RuleRunner) -> None:
                   tools=["echo"],
                   command="./test.sh msg.txt xyzzy",
                 )
+
+                # Check whether `runnable_dependencies` works.
+                system_binary(
+                    name="sed",
+                    binary_name="sed",
+                )
+                system_binary(
+                    name="test",
+                    binary_name="test",
+                    fingerprint_args=["1", "=", "1"]
+                )
+                experimental_test_shell_command(
+                  name="pass_with_runnable_dependency",
+                  execution_dependencies=[":msg-gen", ":src"],
+                  tools=["echo"],
+                  runnable_dependencies=[":sed", ":test"],
+                  command="value=$(sed -e 's/ss/SS/;' < msg.txt) && test $value = meSSage",
```

Comment:
sed on Mac takes different args. Does that matter here?

Reply:
The test doesn't need to use `sed` per se. I could just have it "read" the input with `cat` and then call `test`.

Reply:
So will change to avoid the platform-specific behavior you have identified!